### PR TITLE
feat: async image state based on uploading list

### PR DESCRIPTION
**Pull request by DiamondThree**

### CODE REVIEW

File: packages/blocks/src/embed-block/image/image-block.ts [hunk 1]
```diff
@@ -117,17 +117,52 @@ export class ImageBlockComponent extends NonShadowLitElement {
...
+    this._uploadingStatus = 'loading';
     // exclude padding and border width
     const { width, height } = this.model;
-    const storage = await this.model.page.blobs;
-    assertExists(storage);
-    const url = await storage.get(this.model.sourceId);
-    url && (this._source = url);
+    if (this.model.page.awarenessAdapter.isUploading(this.model.sourceId)) {
+      this._uploadingStatus = 'loading';
+      this.model.page.awarenessAdapter.signals.update.subscribe(
```
Comment (himself65) [inline packages/blocks/src/embed-block/image/image-block.ts:145]: call the dispose function on `disconnectedCallback `

Comment (DiamondThree) [reply packages/blocks/src/embed-block/image/image-block.ts:145]: ok

File: packages/store/src/awareness.ts [hunk 2]
```diff
@@ -118,6 +119,21 @@ export class AwarenessAdapter<
     }
   };
 
+  public isUploading(id: string): boolean {
+    const uploading = this.getAllUploading();
+    return uploading.includes(id);
+  }
+
+  public addUploading(id: string): void {
```
Comment (himself65) [inline packages/store/src/awareness.ts:127]: where is the function that calls `addUploading`?

**Comment by alt1o** [comment]
#918

**Comment by doodlewind** [comment]
Closing in favor of #918, please let me know if I missed something or there will be more updates here.

**Pull request closed without merging**